src/components/about-us/section-04/index.js

```diff
@@ -1,165 +1,335 @@
-//           <TitleWrapper>
-//             <SectionTitle />
-//           </TitleWrapper>  
-//           <ContentContainer>
-//             <InfoCardsList 
-//               showBillboard={this._showBillboard}
-//             />
-//             <InfoCard 
-//               story={this.state.currentStory} 
-//               showPhoto={this._showPhotoBlock}
-//               seeingMore={this.state.seeingMore}
-//             />
-//             <Map
-//               seeingMore={this.state.seeingMore}
-//               story={this.state.currentStory}
-//               storySelector={this._storySelector}
-//               closePhotoBlock={this._closePhotoBlock}
-//             />
-//           </ContentContainer>
-//         </Container>
-//         <InfoBillboard 
-//           isOpened={this.state.billboardIsOpened} 
-//           closeBillboard={this._closeBillboard} 
-//           story={this.state.currentStory}
-//         />
-//       </React.Fragment>
-//     )
-//   }
-// }
-
-// export default Section4
+export default class Section4 extends PureComponent {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedLogo: null,
+      selectedRow: null,
+      infoPageNum: 0,
+      isBorderBottomfixed: true
+    }
+  }
+  _select = (logoIndex) => {
+    this.setState({ 
+      selectedLogo: logoIndex,
+      selectedRow: Math.floor(logoIndex / column.desktop),
+      infoPageNum: 0
+    })
+  }
+  _getLogoBlockWidthOnDesktop = (logoIndex, selectedLogo, selectedRow) => {
+    let logoRow = Math.floor(logoIndex / column.desktop)
+    if (selectedLogo === logoIndex && selectedRow === logoRow) {
+      return logoBlockWidthOnDesktop
+    } else if (selectedRow === logoRow) {
+      return `calc((100% - ${logoBlockWidthOnDesktop}) / ${column.desktop - 1})`
+    }
+    return `calc(100% / ${column.desktop})`
+  }
+  _nextPage = () => {
+    this.setState({ infoPageNum: this.state.infoPageNum + 1 })
+  }  
+  _closeInfoBox = () => {
+    this.setState({ 
+      selectedLogo: null,
+      selectedRow: null
+    })
+  }
+  _getSelectedContent = () => {
+    let { selectedLogo } = this.state
+    if (selectedLogo === null ) return
+    return groupedContent[content[selectedLogo].partnerId]
+  }
+  _onPositionChange = (prevPos, currPos) => {
+    if (prevPos === 'inside' && currPos === 'below') {
+      this.setState({ isBorderBottomfixed: true })
+    } else if (prevPos === 'below' && currPos === 'inside') {
+      this.setState({ isBorderBottomfixed: false })
+    }
+  }
+  _getBorderZIndex = () => {
+    return defaultZIndex + 1
+  }
+  render() {
+    let { selectedLogo, selectedRow, infoPageNum } = this.state
+    const LogoBlockList = content.map((data, index) => {
+      return (
+        <LogoBlock 
+          key={index} 
+          selectedLogo={selectedLogo}
+          onClick={() => this._select(index)}
+          widthOnDesktop={() => this._getLogoBlockWidthOnDesktop(index, selectedLogo, selectedRow)}
```

Comment:
How about adding animation effects on clicking? 
Wrapped box become wider?
Selected info box shows up smoothly?

Reply:
I will use `requestAnimationFrame` to make the animation (wrapped box become wider when selected) more smoothly.